File: C++/leet_Code_Daily/1800_prblm.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
  int maxAscendingSum(vector<int> &nums)
  {
    int n = nums.size();
    if (n == 1)
      return nums[0];

    int curr_sum = nums[0], max_sum = curr_sum;

    for (int i = 1; i < n; i++)
    {
      if (nums[i] > nums[i - 1])
      {
        curr_sum += nums[i]; // Extend the subarray
      }
      else
      {
        curr_sum = nums[i]; // Start new subarray
      }
      max_sum = max(max_sum, curr_sum);
    }

    return max_sum;
  }
};

int main()
{
  Solution sol;
  vector<int> nums = {12, 17, 15, 13, 10, 11, 12};
  cout << "Maximum Ascending subArray sum: " << sol.maxAscendingSum(nums) << endl;
  return 0;
}

File: leetCodeDaily/JavaScript/Array/subArray/1800_prblm.js
var maxAscendingSum = function (nums) {
  let n = nums.length;
  if (n === 1) return nums[0]; // Handle single-element case

  let curr_sum = nums[0];
  let max_sum = curr_sum;

  for (let i = 1; i < n; i++) {
    if (nums[i] > nums[i - 1]) {
      curr_sum += nums[i]; // Extend the current subarray
    } else {
      curr_sum = nums[i]; // Start a new subarray
    }
    max_sum = Math.max(max_sum, curr_sum);
  }

  return max_sum;
};

// Test case
console.log(maxAscendingSum([12, 17, 15, 13, 10, 11, 12])); // Output: 29
console.log(maxAscendingSum([100, 10, 1])); // Output: 100
console.log(maxAscendingSum([6])); // Output: 6
console.log(maxAscendingSum([12, 17, 17])); // Output: 17
